Objects/typevarobject.c

```diff
@@ -1640,6 +1640,11 @@ int _Py_initialize_generic(PyInterpreterState *interp)
     MAKE_TYPE(paramspecargs);
     MAKE_TYPE(paramspeckwargs);
 #undef MAKE_TYPE
+    if (PyDict_SetItemString(interp->cached_objects.generic_type->tp_dict,
+                             "_is_protocol", Py_False) < 0) {
+        return -1;
+    }
+    PyType_Modified(interp->cached_objects.generic_type);
```

Comment:
The alternative fix would be to do this in `typing.py`:

```diff
diff --git a/Lib/typing.py b/Lib/typing.py
index b32ff0c6ba..d327e69848 100644
--- a/Lib/typing.py
+++ b/Lib/typing.py
@@ -1894,7 +1894,8 @@ def _proto_hook(other):
                     annotations = getattr(base, '__annotations__', {})
                     if (isinstance(annotations, collections.abc.Mapping) and
                             attr in annotations and
-                            issubclass(other, Generic) and other._is_protocol):
+                            issubclass(other, Generic)
+                            and getattr(other, "_is_protocol", False):
                         break
                 else:
                     return NotImplemented
@@ -2059,7 +2060,7 @@ def close(self): ...
     Warning: this will check only the presence of the required methods,
     not their type signatures!
     """
-    if not issubclass(cls, Generic) or not cls._is_protocol:
+    if not issubclass(cls, Generic) or not getattr(other, "_is_protocol", False):
         raise TypeError('@runtime_checkable can be only applied to protocol classes,'
                         ' got %r' % cls)
     cls._is_runtime_protocol = True
```

Point in favour: It would mean a few less lines of C code, and would mean we wouldn't be setting an arguably-unnecessary attribute on all subclasses of `Generic`.

Point against: Third-party libraries might be using the `_is_protocol` attribute (even though they shouldn't be), and your approach would ensure that any third-party users don't get unnecessarily broken.